src/include/parser/create_statement.h

```diff
@@ -48,23 +48,94 @@ struct ColumnDefinition {
     TEXT,
 
     VARCHAR,
-    VARBINARY
+    VARBINARY,
+
+    DATE
   };
 
   ColumnDefinition(DataType type) : type(type) {
     // Set varlen to TEXT_MAX_LENGTH if the data type is TEXT
-    if (type == DataType::TEXT)
-      varlen = type::PELOTON_TEXT_MAX_LEN;
+    if (type == DataType::TEXT) varlen = type::PELOTON_TEXT_MAX_LEN;
   }
 
-  ColumnDefinition(char* name, DataType type) : name(name), type(type) {
+  ColumnDefinition(char *name, DataType type) : name(name), type(type) {
     // Set varlen to TEXT_MAX_LENGTH if the data type is TEXT
-    if (type == DataType::TEXT)
-      varlen = type::PELOTON_TEXT_MAX_LEN;
+    if (type == DataType::TEXT) varlen = type::PELOTON_TEXT_MAX_LEN;
   }
 
   virtual ~ColumnDefinition() {}
 
+  static DataType StrToDataType(char *str) {
+    DataType data_type;
+    // Transform column type
+    if ((strcmp(str, "int") == 0) || (strcmp(str, "int4") == 0)) {
+      data_type = ColumnDefinition::DataType::INT;
+    } else if (strcmp(str, "varchar") == 0) {
+      data_type = ColumnDefinition::DataType::VARCHAR;
+    } else if (strcmp(str, "int8") == 0) {
+      data_type = ColumnDefinition::DataType::BIGINT;
+    } else if (strcmp(str, "int2") == 0) {
+      data_type = ColumnDefinition::DataType::SMALLINT;
+    } else if (strcmp(str, "timestamp") == 0) {
+      data_type = ColumnDefinition::DataType::TIMESTAMP;
+    } else if (strcmp(str, "bool") == 0) {
+      data_type = ColumnDefinition::DataType::BOOLEAN;
+    } else if (strcmp(str, "bpchar") == 0) {
+      data_type = ColumnDefinition::DataType::CHAR;
+    } else if ((strcmp(str, "double") == 0) || (strcmp(str, "float8") == 0)) {
+      data_type = ColumnDefinition::DataType::DOUBLE;
+    } else if ((strcmp(str, "real") == 0) || (strcmp(str, "float4") == 0)) {
+      data_type = ColumnDefinition::DataType::FLOAT;
+    } else if (strcmp(str, "numeric") == 0) {
+      data_type = ColumnDefinition::DataType::DECIMAL;
+    } else if (strcmp(str, "text") == 0) {
+      data_type = ColumnDefinition::DataType::TEXT;
+    } else if (strcmp(str, "tinyint") == 0) {
+      data_type = ColumnDefinition::DataType::TINYINT;
+    } else if (strcmp(str, "varbinary") == 0) {
+      data_type = ColumnDefinition::DataType::VARBINARY;
+    } else if (strcmp(str, "date") == 0) {
+      data_type = ColumnDefinition::DataType::DATE;
+    } else {
+      throw NotImplementedException(
+          StringUtil::Format("DataType %s not supported yet...\n", str));
+    }
+    return data_type;
+  }
+
+  static type::TypeId StrToValueType(char *str) {
```

Comment:
How are `StrToDataType` and `StrToValueType` different? They look really similar to me?

Reply:
StrToDataType is for transforming to DataType, StrToValueType is for transforming typeId.